python/graphstorm/model/lm_embed.py

```diff
@@ -327,25 +340,28 @@ def update_cache(self, lm_infer_batch_size, use_fp16=True):
         lm_infer_batch_size: int
             Language model inference batch size
         use_fp16 : bool
-            Use float16 to store BERT embeddings.
+            Use float16 to store LM embeddings.
+
+        Returns
+        -------
+        bool : return True if new LM embeddings are computed else return False.
         """
-        # If the embeddings have been cached, we just load them instead of
+        # If the embeddings have been cached in files, we should load them instead of
         # computing them from scratch.
-        if self._embed_path is not None:
+        if self._embed_path is not None and len(self._lm_emb_cache) == 0:
             self._load_embeddings()
 
-        # If all embeddings are cached:
+        # If all cached embeddings has been loaded to memory:
         if np.all([ntype in self._lm_emb_cache for ntype in self._lm_models.ntypes]):
-            # check if lm is updated, if lm is updated, clear the current cache
+            # check if lm is updated. if it is updated, clear the current cache
             if self._lm_hash != self._lm_models.get_all_lm_hashes():
                 self._clear_cache()
             else:
                 # if lm models was not updated, don't compute the embeddings again
-                return
+                return False
```

Comment:
Who is consuming this return value?

Reply:
i think this is mainly used for testing right now. otherwise, it's hard to tell if the embeddings are computed from scratch or are loaded from files.

Reply:
OK